packages/terra-form-select/src/Select.jsx

```diff
@@ -0,0 +1,297 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames/bind';
+import Popup from 'terra-popup';
+import Arrange from 'terra-arrange';
+import IconCaretDown from 'terra-icon/lib/icon/IconCaretDown';
+import 'terra-base/lib/baseStyles';
+import SelectOption from './SelectOption';
+import SelectMenu from './_SelectMenu';
+import SelectUtils from './_SelectUtils';
+import styles from './Select.scss';
+
+const cx = classNames.bind(styles);
+
+const propTypes = {
+  /**
+   * Function to trigger when the user changes the select value.
+   */
+  onChange: PropTypes.func,
+
+  /**
+   * Name of the select field.
+   */
+  name: PropTypes.string,
+
+  /**
+   * Whether the select is required or not.
+   */
+  required: PropTypes.bool,
+
+  /**
+   * The value to start the select on.
+   */
+  defaultValue: PropTypes.string,
+
+  /**
+   * The value of the select element. Use this to create a controlled input.
+   */
+  value: PropTypes.string,
+
+  /**
+   * Select options
+   */
+  children: PropTypes.node.isRequired,
+
+  /**
+   * Indicates if the select is in an invalid state.
+   */
+  isInvalid: PropTypes.bool,
+
+  /**
+   * Indicates if the select should be disabled.
+   */
+  disabled: PropTypes.bool,
+
+  /**
+   * A callback function to let the containing component (e.g. modal) to regain focus.
+   */
+  releaseFocus: PropTypes.func,
+
+  /**
+   * A callback function to request focus from the containing component (e.g. modal).
+   */
+  requestFocus: PropTypes.func,
+};
+
+const defaultProps = {
```

Comment:
Do we need to default onChange, requestFocus, and releaseFocus to undefined?

Reply:
Not really. I prefer not to set a defaultProp if it's going to be undefined since that's what it would be anyway if the consumer doesn't pass a value.